src/vellum/plugins/vellum_mypy.py

```diff
@@ -644,6 +657,24 @@ def _resolve_descriptors_in_outputs(self, type_info: SymbolTableNode) -> SymbolT
 
         return new_type_info
 
+    def get_type_analyze_hook(self, fullname: str) -> Optional[Callable[[AttributeContext], MypyType]]:
+        if fullname == "builtins.dict":
+            return self._analyze_mapping_type
+        return None
```

Comment:
this could be dangerous bc we are amending _all_ dicts, instead of just ones defined on Nodes